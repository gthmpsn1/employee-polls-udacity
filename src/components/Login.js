import { useState } from "react";
import { connect } from "react-redux";
import {useNavigate} from "react-router-dom";
import {setAuthedUser} from "../actions/authedUser";

const Login = (props) => {
    const navigate = useNavigate();
    const [userNameInput, setUserNameInput] = useState("");
    const [passwordInput, setPasswordInput] = useState("");
    const {users, dispatch} = props;

    const setUNameQuery = (e) => {
        e.preventDefault();
        setUserNameInput(e.target.value);
    }

    const setPasswordQuery = (e) => {
        e.preventDefault();
        setPasswordInput(e.target.value);
    }

    const attemptSignIn = (e) => {
        e.preventDefault();
        if (userNameInput in users) {
            dispatch(setAuthedUser(userNameInput));
            setPasswordInput("");
            setUserNameInput("");
            navigate("/");
        } else {
            alert("Enter valid login information");
        }                
    }


    return (
        <div className="login-container">
            <h1>Employee Polls</h1>
            <div className="login-image"></div>
            <h4>Log In</h4>
            <form className="login-form">
                <label>User</label>
                <input data-testid={'test-authed-user'} type="text" placeholder="User" value={userNameInput} onChange={setUNameQuery} />
                <label>Password</label>
                <input data-testid={'test-password'} type="password" placeholder="Password" value={passwordInput} onChange={setPasswordQuery} />
                <input data-testid={'test-login-button'} className="submit-btn" type="submit" onClick={attemptSignIn} disabled={userNameInput === "" || passwordInput === ""} />
            </form>
        </div>
    )
}
const mapStateToProps = ({users, dispatch}) => ({
    dispatch: dispatch,
    users: users
})

export default connect(mapStateToProps)(Login);