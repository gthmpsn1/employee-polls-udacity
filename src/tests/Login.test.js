/**
 * @jest-environment jsdom
 */

import { render, screen, fireEvent } from "@testing-library/react";
import middleware from "../middleware";
import reducer from "../reducers";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import Login from "../components/Login";

describe("Render Login", () => {
    let mockStore = "";
    let view = "";    

    beforeEach(() => {
        mockStore = createStore(reducer, middleware);
        view = render(
            <MemoryRouter>
                <Provider store={ mockStore }>
                    <Login />
                </Provider>
            </MemoryRouter>
        );
    });

    it("will render login component", () => {
        expect(view).toMatchSnapshot();
    });
});