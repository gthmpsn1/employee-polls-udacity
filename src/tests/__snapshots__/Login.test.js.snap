// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login will render login component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-container"
      >
        <h1>
          Employee Polls
        </h1>
        <div
          class="login-image"
        />
        <h4>
          Log In
        </h4>
        <form
          class="login-form"
        >
          <label>
            User
          </label>
          <input
            data-testid="test-authed-user"
            placeholder="User"
            type="text"
            value=""
          />
          <label>
            Password
          </label>
          <input
            data-testid="test-password"
            placeholder="Password"
            type="password"
            value=""
          />
          <input
            class="submit-btn"
            data-testid="test-login-button"
            disabled=""
            type="submit"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-container"
    >
      <h1>
        Employee Polls
      </h1>
      <div
        class="login-image"
      />
      <h4>
        Log In
      </h4>
      <form
        class="login-form"
      >
        <label>
          User
        </label>
        <input
          data-testid="test-authed-user"
          placeholder="User"
          type="text"
          value=""
        />
        <label>
          Password
        </label>
        <input
          data-testid="test-password"
          placeholder="Password"
          type="password"
          value=""
        />
        <input
          class="submit-btn"
          data-testid="test-login-button"
          disabled=""
          type="submit"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
